kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/clients/PythonClient.java

```diff
@@ -0,0 +1,133 @@
+        String name = Constants.KAFKA_PRODUCER_CLIENT_LABEL + "-python-" + TestUtils.getRandomPodNameSuffix();
+        String jsonOverrides = KubeUtils.isOcp() ? TestUtils.getJsonFileContent("nonJVMClient_openshift.json").replace("%NAME%", name) : "";
+
+        List<String> executableCommand = new ArrayList<>(List.of(cmdKubeClient(deployNamespace).toString(), "run", "-i",
+                "-n", deployNamespace, name,
+                "--image=" + Constants.PYTHON_CLIENT_IMAGE,
+                "--override-type=strategic",
+                "--overrides=" + jsonOverrides,
+                "--", "python3", "/usr/src/confluent-kafka-python/Producer.py", "-b", bootstrap, "-t", topicName));
+        recordKey.ifPresent(key -> {
+            executableCommand.add("-k");
+            executableCommand.add(key);
+        });
+
+        KafkaUtils.produceMessagesWithCmd(deployNamespace, executableCommand, String.valueOf(msg), name, KafkaClientType.PYTHON.name().toLowerCase());
+    }
+
+    @Override
+    public List<ConsumerRecord> consumeMessages(String topicName, String bootstrap, int numOfMessages, Duration timeout) {
+        LOGGER.atInfo().log("Consuming messages using python");
+        String name = Constants.KAFKA_CONSUMER_CLIENT_LABEL + "-python-" + TestUtils.getRandomPodNameSuffix();
+        // Running consumer with parameters to get the latest N number of messages received to avoid consuming twice the same messages
+        List<String> args = List.of("python3", "/usr/src/confluent-kafka-python/Consumer.py", "-n", String.valueOf(numOfMessages), "-b", bootstrap, "-t", topicName);
+        Job pythonClientJob = TestClientsJobTemplates.defaultPythonJob(name, args).build();
+        String podName = KafkaUtils.createJob(deployNamespace, name, pythonClientJob);
+        String log = waitForConsumer(deployNamespace, podName, timeout);
+        KafkaUtils.deleteJob(pythonClientJob);
+        LOGGER.atInfo().setMessage("Log: {}").addArgument(log).log();
+        Stream<String> logRecords = extractRecordLinesFromLog(log);
+        return getConsumerRecords(logRecords);
+    }
+
+    private String waitForConsumer(String namespace, String podName, Duration timeout) {
+        DeploymentUtils.waitForPodRunSucceeded(namespace, podName, timeout);
+        return kubeClient().logsInSpecificNamespace(namespace, podName);
+    }
+
+    private Stream<String> extractRecordLinesFromLog(String log) {
+        return Stream.of(log.split("\n"))
+                .filter(l -> l.contains(RECEIVED_MESSAGE_MARKER))
+                .map(line -> {
+                    final String[] split = line.split(RECEIVED_MESSAGE_MARKER, 2); // Limit is 1 based so a limit of 2 means use the seek at most 1 times
+                    if (split.length > 1) {
+                        return split[1];
+                    }
+                    else {
+                        return "";
+                    }
```

Comment:
I worry this is going to truncate any message that spans multiple lines. e.g. if the message was 

```
hello Sam! 

This is a Kafka record
```

This will produce a single record of `hello Sam!` and loose the `This is a Kafka record``

We need to buffer lines between record markers and then join them.

Reply:
Oh right the strings are json, can we rely on that not containing `\n`?